### PR TITLE
Unet fix

### PR DESCRIPTION
**Pull request by canberk17**
# What does this PR do?

I was doing the CustomDiffusion tutorial, and I started receiving an `UnboundLocalError` at the end of the training.

```
File "/content/diffusers/examples/custom_diffusion/train_custom_diffusion.py", line 1330, in <module>
    main(args)
  File "/content/diffusers/examples/custom_diffusion/train_custom_diffusion.py", line 1312, in main
    images=images,
UnboundLocalError: local variable 'images' referenced before assignment
```

The line (1312) is referring inside the conditional block `if args.push_to_hub:`.  This error error occurs because the `images` variable is defined inside two nested conditional blocks (lines 1232 and 1290), but is later referenced without being assigned if certain conditions are not met.

Here's how `images` is defined:

**line 1232**
```
if accelerator.is_main_process:
    if args.validation_prompt is not None and global_step % args.validation_steps == 0:
        # ...
```
**line 1290**
```
if args.validation_prompt and args.num_validation_images > 0:
    images = [
        # ...
    ]
```

This issue arises if you try to push your model to the hub without, setting the **num_validation_images**, and it will  lead to a "UnboundLocalError" when the code tries to access `images` without having defined it first.

To ensure that the images variable is always initialized, we can declare an empty list at the beginning of the code block where the main process is checked. This will prevent the UnboundLocalError when you reference images when pushing your model to the hub, and don't define the **num_validation_images**

Referring to these: #4704 #4707 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [] Did you write any new necessary tests?

## Who can review?

@sayakpaul @patrickvonplaten

### CODE REVIEW

File: src/diffusers/models/unet_2d_condition.py
Comment (sayakpaul) [inline src/diffusers/models/unet_2d_condition.py:815]: Irrelevant change. Please discrd.

**Review by sayakpaul — APPROVED** [review]
Thanks!

File: src/diffusers/models/unet_2d_condition.py
Comment (canberk17) [reply src/diffusers/models/unet_2d_condition.py]: done

**Comment by sayakpaul** [comment]
Will merge once the CI is green :-) 

Thanks a lot for spotting out the bug and for your contribution.

**Review by patrickvonplaten — APPROVED** [review]

**Comment by canberk17** [comment]
Would you like me to apply the black code formatter, and push it again?

**Comment by sayakpaul** [comment]
Once the CI is green, we can merge :-)

**Pull request merged**
